firmware/drivers/citiroc/citiroc.c

```diff
@@ -87,6 +87,23 @@ void citiroc_daq_set_hvps_curr(uint16_t curr)
 	CITIROC->HVPSR |= ((curr & 0xffff)<<16);
 }
 
+void citiroc_calib_set(uint32_t calibration)
+{
+	uint32_t temp;
+	temp = calibration & CALIBPER;
+
+	/*For period greater than zero, apply the register, else, disable pulse*/
+	if (temp != 0)
+	{
+		CITIROC->ASICCALR &= ~(0xffffffff);
+		CITIROC->ASICCALR |= (calibration & 0xffffffff);
```

Comment:
Neither the OR nor the AND operations are needed on this line. Since you are clearing the register above, you can simply assign the new value to the input parameter.